### PR TITLE
Add caching support for persist Redis

### PR DESCRIPTION
**Pull request by dinukaamarasinghe817**
## Purpose
To support project-level caching for Redis DB in `bal persist`

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6252

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/Module.md [hunk 1]
```diff
@@ -62,17 +62,22 @@ You need to set a value for the following configuration parameter in the `Config
...
 
 Alternatively, you can provide connection parameters instead of the `URI` as follows.
 ```toml
-[<packageName>.<moduleName>.redis.connection]
+[<packageName>.<moduleName>.connectionConfig.connection]
 host = "localhost"
 port = 6379
 ```
 
+Currently, the Redis data store only supports application-level caching. The `maxAge` parameter in the cache configuration specifies the maximum age in `seconds` for cached objects. A value of `-1` indicates that objects within the cache have no expiry time, persisting indefinitely until manually removed.
```
Comment (daneshk) [inline ballerina/Module.md:79]:
```suggestion
Currently, the Redis data store only supports client-level global caching. The `maxAge` parameter in the cache configuration specifies the maximum age in `seconds` for cached objects. A value of `-1` indicates that objects within the cache have no expiry time, persisting indefinitely until manually removed.
```

File: ballerina/Package.md [hunk 2]
```diff
@@ -62,17 +62,22 @@ You need to set a value for the following configuration parameter in the `Config
...
 
 Alternatively, you can provide connection parameters instead of the `URI` as follows.
 ```toml
-[<packageName>.<moduleName>.redis.connection]
+[<packageName>.<moduleName>.connectionConfig.connection]
 host = "localhost"
 port = 6379
 ```
 
+Currently, the Redis data store only supports application-level caching. The `maxAge` parameter in the cache configuration specifies the maximum age in `seconds` for cached objects. A value of `-1` indicates that objects within the cache have no expiry time, persisting indefinitely until manually removed.
```
Comment (daneshk) [inline ballerina/Package.md:79]:
```suggestion
Currently, the Redis data store only supports global cache configuration at the client level. The `maxAge` parameter in the cache configuration specifies the maximum age in `seconds` for cached objects. A value of `-1` indicates that objects within the cache have no expiry time, persisting indefinitely until manually removed.
```

File: ballerina/redis_client.bal [hunk 3]
```diff
@@ -411,6 +430,10 @@ public isolated client class RedisClient {
         }
     }
 
+    private isolated function isCache() returns boolean {
```
[the remark this answers is not in the document]
Comment (dinukaamarasinghe817) [reply ballerina/redis_client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerinax-persist.redis/pull/10/commits/72ac6cb961f422af8a7447db9bd91084c7579d4c

File: ballerina/Package.md (hunk 2 above)
Comment (dinukaamarasinghe817) [reply ballerina/Package.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-persist.redis/pull/10/commits/72ac6cb961f422af8a7447db9bd91084c7579d4c

File: ballerina/Module.md (hunk 1 above)
Comment (dinukaamarasinghe817) [reply ballerina/Module.md]: Fixed in https://github.com/ballerina-platform/module-ballerinax-persist.redis/pull/10/commits/72ac6cb961f422af8a7447db9bd91084c7579d4c

**Review by daneshk — APPROVED** [review]

**Pull request merged**